### PR TITLE
Docker support for Rabbit cluster

### PR DESCRIPTION
**Pull request by pupapaik**
Role independent for rabbitmq-cluster

### CODE REVIEW

File: rabbitmq/cluster.sls [hunk 1]
```diff
@@ -5,7 +5,7 @@ include:
 - linux.network.host
 - rabbitmq.server.service
 
-{% if cluster.get('role', 'slave') == 'master' %}
+{% if cluster.get('role', 'none') == 'master' %}
```
Comment (fpytloun) [inline rabbitmq/cluster.sls]: Use pythonish None instead of string.

File: rabbitmq/files/entrypoint.sh [hunk 2]
```diff
@@ -1,6 +1,29 @@
-{%- from "rabbitmq/map.jinja" import server with context -%}
+{%- from "rabbitmq/map.jinja" import server,cluster with context -%}
 #!/bin/bash -e
 
+{%- if cluster.enabled %}
```
Comment (fpytloun) [inline rabbitmq/files/entrypoint.sh]: Better use `cluster.get('enabled', False)`

File: rabbitmq/files/entrypoint.sh [hunk 3]
```diff
@@ -1,6 +1,29 @@
...
+
+# if NUM_NODES not passed, default to 3
+if [ -z "$NUM_NODES" ]; then
+  NUM_NODES=3
+fi
+
+# loop through number of nodes
+for NUM in `seq 1 $NUM_NODES`; do
```
Comment (fpytloun) [inline rabbitmq/files/entrypoint.sh]: 1. use lower-case for local variables during iteration
2. don't need to exec seq here
   `for num in 1..${NUM_NODES}`

File: rabbitmq/files/entrypoint.sh [hunk 4]
```diff
@@ -1,6 +1,29 @@
...
+  NODE_SERVICE_HOST="RABBITMQ_SERVER_NODE0${NUM}_SERVICE_HOST"
+
+  # if set
+  if [ -n "${!NODE_SERVICE_HOST}" ]; then
```
Comment (fpytloun) [inline rabbitmq/files/entrypoint.sh:17]: This doesn't make sense, you define this variable above.

File: rabbitmq/files/entrypoint.sh [hunk 5]
```diff
@@ -1,6 +1,29 @@
...
+    # if not its own IP, then add it
+    if [ $(expr "$HOSTNAME" : "rabbitmq-server-node0${NUM}") -eq 0 ]; then
```
Comment (fpytloun) [inline rabbitmq/files/entrypoint.sh]: I wouldn't force user to match rabbitmq-server-node pattern of hostname.
Try this pattern:

```
if [[ "$HOSTNAME" =~ ^.+01$ ]]; then
    # master
```

File: rabbitmq/files/entrypoint.sh (hunk 4 above)
Comment (pupapaik) [inline rabbitmq/files/entrypoint.sh:17]: I am reusing same approach from official kubernetes repository for galera.

File: rabbitmq/files/entrypoint.sh [hunk 6]
```diff
@@ -1,6 +1,29 @@
-{%- from "rabbitmq/map.jinja" import server with context -%}
+{%- from "rabbitmq/map.jinja" import server,cluster with context -%}
 #!/bin/bash -e
 
+{%- if cluster.enabled %}
+
+# if NUM_NODES not passed, default to 3
+if [ -z "$NUM_NODES" ]; then
+  NUM_NODES=3
+fi
```
Comment (fpytloun) [inline rabbitmq/files/entrypoint.sh]: Use much simpler syntax:
NUM_NODES=${NUM_NODES:-3}

without need for if conditional

File: rabbitmq/files/entrypoint.sh [hunk 7]
```diff
@@ -1,6 +1,30 @@
...
 #!/bin/bash -e
 
+{%- if cluster.get('enabled', False) %}
+
+# if NUM_NODES not passed, default to 3
+NUM_NODES=${NUM_NODES:-3}
+
+# loop through number of nodes
+for num in 1..${NUM_NODES}; do
+  NODE_SERVICE_HOST="RABBITMQ_SERVER_NODE0${num}_SERVICE_HOST"
```
Comment (fpytloun) [inline rabbitmq/files/entrypoint.sh:14]: What if num will be 10 or more? :-)

File: rabbitmq/files/entrypoint.sh [hunk 8]
```diff
@@ -1,6 +1,30 @@
...
+
+  # if set
+  if [ -n "${!NODE_SERVICE_HOST}" ]; then
```
Comment (fpytloun) [inline rabbitmq/files/entrypoint.sh:17]: This is not needed, remove this conditional

File: rabbitmq/files/entrypoint.sh [hunk 9]
```diff
@@ -1,6 +1,30 @@
...
+    # if not its own IP, then add it
+    if [ $(expr "$HOSTNAME" : "rabbitmq-server-node0${num}") -eq 0 ]; then
```
Comment (fpytloun) [inline rabbitmq/files/entrypoint.sh:19]: Update as I mentioned before

File: rabbitmq/files/entrypoint.sh [hunk 10]
```diff
@@ -1,6 +1,30 @@
...
+      # set role cluster role slave for salt
+      sed -i -e "s|role: none|role: slave|" /srv/salt/pillar/rabbitmq-server.sls
```
Comment (fpytloun) [inline rabbitmq/files/entrypoint.sh]: maybe use better expression for matching none: `role:\ [Nn]one`
`'s,^\([\ \t]*\)role:\ [Nn]one,\1slave,g'`

File: rabbitmq/files/entrypoint.sh (hunk 7 above)
Comment (pupapaik) [inline rabbitmq/files/entrypoint.sh:14]: this is impossible :)

File: rabbitmq/files/entrypoint.sh (hunk 8 above)
Comment (pupapaik) [inline rabbitmq/files/entrypoint.sh:17]: yes, it is. Because you start it one by one. So this is check.

**Comment by pupapaik** [comment]
Verified and tested. I want to merge it! :)

**Comment by mceloud** [comment]
LGTM :+1:

**Pull request merged**